sdk/textanalytics/ai-text-analytics/src/cognitiveServicesCredential.ts

```diff
@@ -1,24 +1,53 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-import { ApiKeyCredentials } from "@azure/core-http";
+import { ServiceClientCredentials, WebResource } from "@azure/core-http";
+
+const SUBSCRIPTION_KEY_HEADER_NAME = "Ocp-Apim-Subscription-Key";
 
 /**
- * Creates a new CognitiveServicesCredential object.
- *
- * @param {string} subscriptionKey   The CognitiveServices subscription key
+ * This class is used to authenticate to Cognitive Services using a Subscription Key retrieved
+ * from the Azure portal.
  */
-export class CognitiveServicesCredential extends ApiKeyCredentials {
+export class CognitiveServicesCredential implements ServiceClientCredentials {
+  /**
+   * Used to authenticate to Cognitive Services
+   */
+  private subscriptionKey!: string;
+
+  /**
+   * Creates a new CognitiveServicesCredential object.
+   *
+   * @param {string} subscriptionKey   The Cognitive Services subscription key for authentication.
+   */
   constructor(subscriptionKey: string) {
-    if (!subscriptionKey || (subscriptionKey && typeof subscriptionKey.valueOf() !== "string")) {
-      throw new Error("subscriptionKey cannot be null or undefined and must be of type string.");
+    this.setSubscriptionKey(subscriptionKey);
+  }
+
+  /**
+   * Updates the Subscription Key used for authentication. Use this method to roll credentials.
+   * @param subscriptionKey The Cognitive Services subscription key for authentication.
+   */
+  public setSubscriptionKey(subscriptionKey: string): void {
+    if (!subscriptionKey || typeof subscriptionKey !== "string") {
+      throw new Error("subscriptionKey must be a non-empty string");
+    }
+
+    this.subscriptionKey = subscriptionKey;
+  }
+
+  /**
+   * Signs a request with the provided Subscription Key.
+   *
+   * @param {WebResource} webResource The WebResource to be signed.
+   * @returns {Promise<WebResource>} The signed request object.
+   */
+  public async signRequest(webResource: WebResource): Promise<WebResource> {
+    if (!webResource) {
```

Comment:
In `ApiKeyCredential.signRequest` there is a check to see if `webResource.headers` exists before setting the value of the auth header. Are we sure we don't need to do that here?

Reply:
I looked into the implementation of `WebRequest` and `headers` is always initialized in the constructor. I think `ApiKeyCredential` is getting too far into defensive coding here, especially for internal APIs protected by TS.